fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/materialization/ColumnRewriter.java

```diff
@@ -0,0 +1,140 @@
+
+import com.google.common.collect.Lists;
+import com.starrocks.sql.optimizer.base.ColumnRefFactory;
+import com.starrocks.sql.optimizer.base.EquivalenceClasses;
+import com.starrocks.sql.optimizer.operator.scalar.ColumnRefOperator;
+import com.starrocks.sql.optimizer.operator.scalar.ScalarOperator;
+import com.starrocks.sql.optimizer.operator.scalar.ScalarOperatorVisitor;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+public class ColumnRewriter {
+    private static final Logger LOG = LogManager.getLogger(ColumnRewriter.class);
+    private RewriteContext rewriteContext;
+
+    public ColumnRewriter(RewriteContext rewriteContext) {
+        this.rewriteContext = rewriteContext;
+    }
+
+    public ScalarOperator rewriteByQueryEc(ScalarOperator predicate) {
+        ColumnRewriteVisitor visitor = new ColumnRewriteVisitor(rewriteContext, false, false, true, true);
+        return predicate.accept(visitor, null);
+    }
+
+    public ScalarOperator rewriteByViewEc(ScalarOperator predicate) {
+        ColumnRewriteVisitor visitor = new ColumnRewriteVisitor(rewriteContext, false, false, true, false);
+        return predicate.accept(visitor, null);
+    }
+
+    public ScalarOperator rewriteViewToQuery(ScalarOperator predicate) {
+        ColumnRewriteVisitor visitor = new ColumnRewriteVisitor(rewriteContext, true, true, false, false);
+        return predicate.accept(visitor, null);
+    }
+
+    public ScalarOperator rewriteViewToQueryWithQueryEc(ScalarOperator predicate) {
+        ColumnRewriteVisitor visitor = new ColumnRewriteVisitor(rewriteContext, true, true, true, true);
+        return predicate.accept(visitor, null);
+    }
+
+    public ScalarOperator rewriteViewToQueryWithViewEc(ScalarOperator predicate) {
+        ColumnRewriteVisitor visitor = new ColumnRewriteVisitor(rewriteContext, true, true, true, false);
+        return predicate.accept(visitor, null);
+    }
+
+    public ScalarOperator rewriteQueryToView(ScalarOperator predicate) {
+        ColumnRewriteVisitor visitor = new ColumnRewriteVisitor(rewriteContext, true, false, false, false);
+        return predicate.accept(visitor, null);
+    }
+
+    private class ColumnRewriteVisitor extends ScalarOperatorVisitor<ScalarOperator, Void> {
+        private boolean enableRelationRewrite;
+        private boolean enableEcRewrite;
+
+        private Map<Integer, Integer> srcToDstRelationIdMapping;
+        private ColumnRefFactory srcRefFactory;
+        private ColumnRefFactory dstRefFactory;
+        private Map<Integer, List<ColumnRefOperator>> srcRelationIdToColumns;
+        private Map<Integer, List<ColumnRefOperator>> dstRelationIdToColumns;
+        private EquivalenceClasses ec;
+
+        public ColumnRewriteVisitor(RewriteContext rewriteContext, boolean enableRelationRewrite, boolean viewToQuery,
```

Comment:
Using Ec as the abbreviation of EquivalenceClasses isn't clear.

Reply:
EquivalenceClasses is too long, so here I use ec.

Reply:
use long name